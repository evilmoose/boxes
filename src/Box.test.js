import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Box from './Box'; // Adjust the import path according to your file structure

describe('Box', () => {
  it('renders without crashing', () => {
    const handleRemove = jest.fn();
    const { getByText } = render(<Box id="test-box" handleRemove={handleRemove} />);

    // Check if the button renders correctly
    // eslint-disable-next-line testing-library/prefer-screen-queries
    expect(getByText(/remove the box!/i)).toBeInTheDocument();
  });

  it('calls handleRemove on button click', () => {
    const handleRemove = jest.fn();
    const { getByText } = render(<Box id="test-box" handleRemove={handleRemove} />);
    
    // Simulate button click
    // eslint-disable-next-line testing-library/prefer-screen-queries
    fireEvent.click(getByText(/remove the box!/i));

    // Check if handleRemove was called
    expect(handleRemove).toHaveBeenCalledWith("test-box");
  });

  it('renders correctly', () => {
    const handleRemove = jest.fn(); // Mock function for handleRemove prop
    const tree = renderer
      .create(<Box id="test-box" handleRemove={handleRemove} width={5} height={5} backgroundColor="blue" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
