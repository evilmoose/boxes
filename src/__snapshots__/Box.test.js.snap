// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "height": "5em",
        "width": "5em",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Remove The Box!
  </button>
</div>
`;
